torch/_inductor/decomposition.py

```diff
@@ -357,10 +358,29 @@ def all_dim(input, dim, keepdim=False):
 @register_decomposition(aten.copy)
 def copy(self, src, non_blocking=False):
     intermediate = src.to(self, non_blocking)
-    if self.size() != intermediate.size():
-        return aten.expand_copy.default(intermediate, self.size())
-    else:
+    # cheapest case
+    if self.size() == intermediate.size() and self.stride() == intermediate.stride():
         return intermediate
+    # next cheapest case
+    if self.size() == intermediate.size() and self.is_contiguous():
+        return intermediate.contiguous()
```

Comment:
It is ok to return `src` as-is here?

Reply:
I added this case for if `self` is contiguous, but `src` isn't. So I don't think returning `src` as-is here is ok.

Reply:
Sorry I meant in the sense of "native functions should never return their input as-is and at least do a self.alias()".